mobile/library/objective-c/EnvoyConfiguration.mm

```diff
@@ -244,4 +285,18 @@ - (instancetype)initWithAdminInterfaceEnabled:(BOOL)adminInterfaceEnabled
   }
 }
 
+// This is only for test purposes
+- (NSString *)generateYamlString {
+  try {
+    Envoy::Platform::EngineBuilder builder = [self applyToCXXBuilder];
+    std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> bootstrap =
+        builder.generateBootstrap();
+    std::string bootstrap_str = Envoy::MessageUtil::getYamlStringFromMessage(*bootstrap);
+    return [NSString stringWithUTF8String:bootstrap_str.c_str()];
+  } catch (const std::exception &e) {
+    NSLog(@"[Envoy] error generating yaml: %@", @(e.what()));
+    return nullptr;
```

Comment:
Added a helper function to the EnvoyConfiguration object to get a yaml string out for tests to parse.  This kinda matches how things work in the jni_interface. Let me know if there's some way to make it test-only that I should do.

Reply:
I assume this function and the Swift one are meant to be analogs for https://github.com/envoyproxy/envoy/blob/3eab4eab695bd7069e7a9a6b2ed83edb3804010f/mobile/library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java#L242 ?

If that's the case, let's put TODOs to move them to test only libraries.  Or we could just move these functions here and the Swift one into separate test directory files and use them in the tests only.